apps/web/src/server/applications/case/representations/representation/attachment-upload/attachment-upload.controller.js

```diff
@@ -1,19 +1,19 @@
-import { getRepresentationPageUrl } from '../representation.utilities.js';
 import { getCaseFolders } from '../../../documentation/applications-documentation.service.js';
+import { getFinalRepPageUrl } from '../../utils/get-final-rep-page-url.js';
 
 const view = 'applications/representations/representation/attachment-upload.njk';
 
 /**
  *  @type {import('@pins/express').RenderHandler<{}, {}, {}, { repId: string, repType: string }, {caseId: string}>}
  */
-export const getRepresentationAttachmentUpload = async ({ query, params }, res) => {
+export const getRepresentationAttachmentUpload = async ({ params }, res) => {
 	const { caseId } = params;
-	const { repId, repType } = query;
 	const folders = await getCaseFolders(Number(caseId));
 	const folder = folders.find((el) => el.displayNameEn === 'Relevant representations');
+	const { locals } = res;
 
 	return res.render(view, {
-		link: getRepresentationPageUrl('check-answers', repId, repType),
+		link: getFinalRepPageUrl(locals.representation, caseId, null),
```

Comment:
i don't think you should pass null as an argument. Have the function default if not set.